keps/sig-node/3953-node-resource-hot-plug/README.md

```diff
@@ -0,0 +1,674 @@
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+This feature is node local and mainly handled in kubelet, It has no dependency on etcd.
+In case there are pending pods and there is hot plug of resources, The scheduler relies on the API server to fetch node information. 
+Without access to the API server, it cannot make scheduling decisions as the node resources are not updated. The pending pods would remain in same condition.
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+This feature mainly does two things fetch machine information from cAdvisor and reinitialize resource managers.
+Failure scenarios can occur in cAdvisor level that is if it wrongly updated with incorrect machine information.
+
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+If enabling this feature causes performance degradation, its suggested not to hot plug resources and restart the kubelet
+to manually to continue operation as before.
+
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+Existing and the alternative to this effort would be restarting the kubelet manually each time after the node resize.
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
```

Comment:
This KEP omits any intent to support hot unplug. But I think this is where we could mention simultaneously adding support for hot unplug and hot plug.

Even when we rule out an alternative thing, we still typically list it as an alternative if it sounds like a viable (less viable, but still feasible) option.

Reply:
Added a Future Work section at the bottom of the KEP or should I cover unplug at some other section?